server/src/service.rs

```diff
@@ -126,18 +128,29 @@ impl Service {
 }
 
 impl Service {
-    pub async fn run() -> Result<()> {
-        let connection = zbus::connection::Builder::session()?
-            .name(oo7::dbus::api::Service::DESTINATION.as_deref().unwrap())?
-            .build()
-            .await?;
-        connection
-            .object_server()
-            .at(
-                oo7::dbus::api::Service::PATH.as_deref().unwrap(),
-                Self::default(),
-            )
+    pub async fn new() -> Result<Self> {
+        Ok(Self {
+            manager: Default::default(),
+            session_index: RwLock::new(0),
+            connection: zbus::connection::Builder::session()?
+                .name(oo7::dbus::api::Service::DESTINATION.as_deref().unwrap())?
+                .build()
+                .await?,
+        })
+    }
+
+    pub async fn run(self) -> Result<()> {
+        let connection = self.connection();
+        let object_server = connection.object_server();
+
+        object_server
+            .at(oo7::dbus::api::Service::PATH.as_deref().unwrap(), self)
             .await?;
+
         Ok(())
     }
+
+    pub fn connection(&self) -> Connection {
```

Comment:
Please either make it so it returns a reference to `Connection` or remove the method and use `let connection = self.conection` in the `run` method. This clone here is a bit unnecesary.

Reply:
Removed the getter method, thanks!